homeassistant/components/vesync/switch.py

```diff
@@ -89,6 +89,10 @@ def update(self):
         self.smartplug.update()
         self.smartplug.update_energy()
 
+    def turn_on(self, **kwargs):
```

Comment:
You could make this more DRY and avoid writing `turn_on` twice by making a 

`class VeSyncBaseSwitch(VeSyncDevice, SwitchEntity)` and having `VeSyncSwitchHA` and `VeSyncLightSwitch` use it as a base class.

Reply:
Makes the most sense to me, done.